crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -2782,13 +2782,18 @@ impl<'src> Parser<'src> {
         // def foo(): ...
         // @@
         // def foo(): ...
+        // @test
         while self.at(TokenKind::At) {
             progress.assert_progressing(self);
 
             let decorator_start = self.node_start();
             self.bump(TokenKind::At);
 
-            let parsed_expr = self.parse_named_expression_or_higher(ExpressionContext::default());
+            let parsed_expr = if self.at(TokenKind::Def) || self.at(TokenKind::Class) {
```

Comment:
This is a somewhat unrelated improvement but it prevents us from parsing `@def` as a decorator calling the function `@def`. I think it's much more likely in this context that `def` is the start of a function.